src/mudlet-lua/lua/TableUtils.lua

```diff
@@ -590,3 +590,38 @@ function table.keys(t)
   end
 	return keys
 end
+
+
+-- Table is_field
+--
+-- @return true or false, indicating if a given field exists inside the given complex table
+---
+--- @usage Example:
+---   <pre>
+---   -- Assume a table like this is received from game via GMCP. For testing, we define here.
+---   local gmcpTable = {
+---     char = {
+---       base = {
+---         info = {
+---           guild = {
+---             foo = "bar"
+---           }
+---         }
+---       }
+---     }
+---   }
+---
+---   if table.is_field(gmcpTable, "char.base.info.guild.foo") then
+---     -- We can now be sure this table exists, and every table inside, until the interesting leaf value 
+---     save_for_later = gmcpTable.char.base.info.guild.foo
+---   end
+---   </pre>
+function table.is_field(t, s)
+  if t == nil then return false end
+  local t = t
```

Comment:
We should make sure that t is a table, and s is a string, and if not raise our own errors (with printError would be good, so we can include stack trace to help them find it) so that it doesn't just throw obscure errors pointing to like 622 of TableUtils.lua. 

Similarly we may want to add some handling around the case where you check for leaf "gmcp.Char.Items.List.list" but "gmcp.Char.Items" turns out to be a string for some reason. It's a bit of extra work now but it'll save us having to look the code up during support for others later.